modules/promotions/assets/js/react/behavior.js

```diff
@@ -0,0 +1,53 @@
+import App from './app';
+import { createRoot } from 'react-dom/client';
+
+export default class ReactPromotionBehavior extends Marionette.Behavior {
+	promotionInfoTip = null;
+
+	ui() {
+		return {
+			animatedHeadlineButton: '[data-promotion].elementor-control-type-switcher',
+		};
+	}
+
+	events() {
+		return {
+			'click @ui.animatedHeadlineButton': 'onClickControlButtonAnimatedHeadline',
+		};
+	}
+
+	onClickControlButtonAnimatedHeadline( event ) {
+		event.stopPropagation();
+		this.mount();
+	}
+
+	mount() {
+		const colorScheme = elementor?.getPreferences?.( 'ui_theme' ) || 'auto',
+			isRTL = elementorCommon.config.isRTL,
+			rootElement = document.querySelector( '.e-promotion-react' ),
+			promotionType = rootElement?.closest( '.elementor-control-type-switcher' )?.dataset?.promotion;
+
+		if ( ! rootElement ) {
+			return;
+		}
+
+		this.promotionInfoTip = createRoot( rootElement );
+
+		this.promotionInfoTip.render(
+			<App
+				colorScheme={ colorScheme }
+				isRTL={ isRTL }
+				promotionsData={ elementorPromotionsData[ promotionType ] }
```

Comment:
@mike-elementor @elchugreeva I have tweaked this a little bit. It now loads the data for the specific promotion type only.

Reply:
I've done that also ^.^)) but it's not enough... also have to sync what we provide from php so that all comes together (from php it goes by a key 'animated-heading' or something)

Reply:
@mike-elementor I did sync the data is JS and PHP, so now it should work.

Reply:
@mike-elementor Apologies for the confusion.  I think you should have less of these issues now that I will take on another task... :)